src/components/auth/SignupForm/index.tsx

```diff
@@ -41,50 +48,115 @@ const SignupForm = () => {
     formState: { isValid },
   } = methods;
 
+  const { isVisible: isSuccessVisible, handleToggleClick: toggleSuccessClick } = useToggleButton();
+  const { isVisible: is400Visible, handleToggleClick: toggle400Click } = useToggleButton();
+  const { isVisible: is409Visible, handleToggleClick: toggle409Click } = useToggleButton();
+
+  const { mutate } = useSignup();
+
+  const onSubmit = (formData: object) => {
+    mutate(formData as SignupParams, {
+      onSuccess: () => {
+        toggleSuccessClick();
+      },
+      onError: (error) => {
+        if ((error as AxiosError)?.response?.status === 400) {
+          toggle400Click();
+        } else if ((error as AxiosError)?.response?.status === 409) {
+          toggle409Click();
+        }
+      },
+    });
+  };
+
```

Comment:
mutate는 어떤 동작을 하는지 명확하게 이름을 변경해주고,  onSubmit은 보내는 역할만 하는거 어떨까요?
onSubmit 함수명도 의미있는 이름으로 변경해도 좋을 것 같습니다!

```suggestion
 const { mutate : signupMutation } = useSignup({
   onSuccess: () => //생략
 });

 const onSubmit = (formData: object) =>{
   signupMutation()
 }
```

Reply:
좋은 의견 감사합니다~ mutate와 onSubmit을 더 직관적으로 수정해볼게요.
onSubmit에서 에러처리까지 하지 않으면 어디서 해야할 지가 애매하네요... 더 고민해보겠습니다!